app/api/suggestions/blankSuggestions.ts

```diff
@@ -2,52 +2,97 @@ import entitiesModel from 'api/entities/entitiesModel';
 import { files } from 'api/files';
 import { EnforcedWithId } from 'api/odm';
 import settings from 'api/settings';
+import { tenants } from 'api/tenants';
 import { propertyTypeIsMultiValued } from 'api/services/informationextraction/getFiles';
 import { ObjectIdSchema } from 'shared/types/commonTypes';
 import { IXExtractorType } from 'shared/types/extractorType';
 import { FileType } from 'shared/types/fileType';
 import { IXSuggestionType } from 'shared/types/suggestionType';
-import { Suggestions } from './suggestions';
 import templates from 'api/templates';
 import { LanguageUtils } from 'shared/language';
+import { Suggestions } from './suggestions';
 
 const fetchEntitiesBatch = async (query: any, limit: number = 100) =>
-  entitiesModel.db.find(query).select('sharedId').limit(limit).sort({ _id: 1 }).lean();
+  entitiesModel.db
+    .find(query)
+    .select(['sharedId', 'language'])
+    .limit(limit)
+    .sort({ _id: 1 })
+    .lean();
 
-const fetchEntitiesSharedIds = async (
+const fetchEntitiesData = async (
   template: ObjectIdSchema,
-  defaultLanguage: string,
+  defaultLanguage?: string,
   batchSize = 2000
 ) => {
   const BATCH_SIZE = batchSize;
   let query: any = {
     template,
-    language: defaultLanguage,
+    ...(defaultLanguage && { language: defaultLanguage }),
   };
 
-  const sharedIdLists: string[][] = [];
+  const dataList: { sharedId: string; language: string }[][] = [];
 
   let fetchedEntities = await fetchEntitiesBatch(query, BATCH_SIZE);
+
   while (fetchedEntities.length) {
-    sharedIdLists.push(fetchedEntities.map(e => e.sharedId!));
+    dataList.push(fetchedEntities.map(e => ({ sharedId: e.sharedId!, language: e.language! })));
     query = {
       ...query,
       _id: { $gt: fetchedEntities[fetchedEntities.length - 1]._id },
+      ...(defaultLanguage && {
+        language: { $gt: fetchedEntities[fetchedEntities.length - 1].language },
+      }),
     };
     // eslint-disable-next-line no-await-in-loop
     fetchedEntities = await fetchEntitiesBatch(query, BATCH_SIZE);
   }
 
-  return sharedIdLists.flat();
+  return dataList.flat();
 };
 
-export const getBlankSuggestion = (
-  file: EnforcedWithId<FileType>,
-  { _id: extractorId, property: propertyName }: { _id: ObjectIdSchema; property: string },
-  template: ObjectIdSchema,
-  propertyType: string,
-  defaultLanguage: string
-) => ({
+const getBlankSuggestionForProperty = ({
+  entityId,
+  extractorId,
+  propertyName,
+  propertyType,
+  template,
+  language,
+}: {
+  entityId: string;
+  extractorId: ObjectIdSchema;
+  propertyName: string;
+  propertyType: string;
+  template: ObjectIdSchema;
+  language: string;
+}) => ({
+  language,
+  entityId,
+  entityTemplate: typeof template === 'string' ? template : template.toString(),
```

Comment:
Again I'm confused by this conditional.  There is a typing on the arguments for template.  If `template` is of type `ObjectIdSchema`, this conditional should not exist.  Is this because the test is passing the wrong thing or can this function really receive both types?  If the former scenario, we should fix the test, if the later, then we either dual-type the property (not ideal) or we ensure that we are passing the right type from all the calls to the function.

I see this is also the case in line 91 and 101 which is unfortunate. :(

Reply:
This was originally as such, but now that we are casting templates as string[], and knowing that those come from the routes (and are therefore strings) we can remove that.